### PR TITLE
EFA-366 Close the sync wizard when handshake was cancelled by the other device

### PR DESCRIPTION
**Pull request by ig-garcia**

### CODE REVIEW

File: k9mail/src/main/java/security/planck/sync/PlanckSyncRepository.kt [hunk 1]
```diff
@@ -98,17 +98,13 @@ class PlanckSyncRepository @Inject constructor(
...
+                cancelIfCancelledByPartner()
             }
 
             SyncHandshakeSignal.SyncNotifyInGroup -> {
                 isGrouped = true
                 k9.markSyncEnabled(true)
-                if (syncState.allowSyncNewDevices) {
-                    startOrResetManualSyncTimer()
-                }
+                cancelIfCancelledByPartner()
```
Comment (Stereodark) [inline k9mail/src/main/java/security/planck/sync/PlanckSyncRepository.kt:107]: should  we also set the isGrouped to false in this case?

Comment (ig-garcia) [reply k9mail/src/main/java/security/planck/sync/PlanckSyncRepository.kt:107]: You mean, in the case of key sync cancelled by partner?

Comment (saschabacardit) [reply k9mail/src/main/java/security/planck/sync/PlanckSyncRepository.kt:107]: cancel should automatically ungroup on the engine, should it not?

Comment (ig-garcia) [reply k9mail/src/main/java/security/planck/sync/PlanckSyncRepository.kt]: no, thats not the use case. Cancel means that it cancels the creation of the new group, but one of the devices could already be in the group and handshaking to add a new device to the same group.

**Review by saschabacardit — APPROVED** [review]

**Pull request merged**